src/cursor/cur_file.c

```diff
@@ -75,6 +75,76 @@ __curfile_equals(WT_CURSOR *a, WT_CURSOR *b, int *equalp)
     API_END_RET(session, ret);
 }
 
+/*
+ * __curfile_set_key --
+ *     WT_CURSOR->set_key implementation for file cursor.
+ */
+static void
+__curfile_set_key(WT_CURSOR *cursor, ...)
+{
+    va_list ap;
+
+    va_start(ap, cursor);
+    F_CLR((WT_CURSOR_BTREE *)cursor, WT_CBT_RESEARCH);
+    WT_IGNORE_RET(__wt_cursor_set_keyv(cursor, cursor->flags, ap));
+    va_end(ap);
+}
+
+/*
+ * __curfile_research --
+ *     Research the saved key to regain the cursor position
+ */
+static int
+__curfile_research(WT_CURSOR *cursor)
```

Comment:
Don't you want to use the cursor's search routine? I'd expect something like
```
ret = cursor->search(cursor);
```
Then you don't need to check `KEY_EXT` or resulting flags, the called function should get that right.

Reply:
No, I need to use the btree level function. Otherwise search will release the page again.